src/components/play/multipleChoices/ChoiceButton.tsx

```diff
@@ -0,0 +1,129 @@
+
+import CheckIcon from '@mui/icons-material/Check';
+import CloseIcon from '@mui/icons-material/Close';
+import { Button, Typography } from '@mui/material';
+
+import { buildMultipleChoicesButtonCy } from '../../../config/selectors';
+import theme from '../../../layout/theme';
+import {
+  ChoiceState,
+  MultipleChoicesChoice,
+  StatusColor,
+} from '../../types/types';
+
+const DEFAULT_COLOR = 'primary';
+const CORRECT_COLOR = 'success';
+const INCORRECT_COLOR = 'error';
+const DEFAULT_DISABLED = 'whitesmoke';
+
+const styleButton = ({
+  color,
+  isSelected,
+  dataCy,
+  endIcon,
+}: {
+  color: StatusColor;
+  isSelected: boolean;
+  dataCy: string;
+  endIcon?: JSX.Element;
+}) => {
+  return {
+    color,
+    variant: isSelected ? 'contained' : 'outlined',
+    endIcon,
+    'data-cy': dataCy,
+  } as const;
+};
+
+const computeDisabledSx = (choiceState: ChoiceState | undefined) => {
+  const successColor = theme.palette.success.main;
+  const errorColor = theme.palette.error.main;
+
+  switch (choiceState) {
+    case ChoiceState.CORRECT:
+      return { backgroundColor: successColor, color: DEFAULT_DISABLED };
+    case ChoiceState.INCORRECT:
+      return { backgroundColor: errorColor, color: DEFAULT_DISABLED };
+    case ChoiceState.MISSING:
+      return { color: successColor, borderColor: successColor };
+    default:
+      return {};
+  }
+};
+
+type Props = {
+  choice: MultipleChoicesChoice;
+  choiceState: ChoiceState;
+  idx: number;
+  isSelected: boolean;
+  isReadonly: boolean;
+  showState: boolean;
+  onClick: (value: string) => void;
+};
+
+export const ChoiceButton = ({
+  choice,
+  choiceState,
+  idx,
+  isSelected,
+  isReadonly,
+  showState,
+  onClick,
+}: Props) => {
+  const computeStyles = () => {
+    let color: StatusColor = DEFAULT_COLOR;
+    let icon;
```

Comment:
To remove the need for mutable variables you can update the switch to a function that directly returns an object with keys `color` and `icon`.